pulgin install requirement

try:
                pip_install = ['uv', 'pip', 'install', '-r', requirements_file]
                if settings.PLUGIN_PIP_CHINA:
                    pip_install.extend(['-i', settings.PLUGIN_PIP_INDEX_URL])

                max_retries = settings.PLUGIN_PIP_MAX_RETRY
                for attempt in range(max_retries):
                    try:
                        subprocess.check_call(
                            pip_install,
                            stdout=subprocess.DEVNULL,
                            stderr=subprocess.DEVNULL,
                        )
                        break
                    except subprocess.TimeoutExpired:
                        if attempt == max_retries - 1:
                            raise PluginInstallError(f'插件 {plugin} 依赖安装超时')
                        continue
                    except subprocess.CalledProcessError as e:
                        if attempt == max_retries - 1:
                            raise PluginInstallError(f'插件 {plugin} 依赖安装失败：{e}') from e
                        continue
            except subprocess.CalledProcessError as e:
                raise PluginInstallError(f'插件 {plugin} 依赖安装失败：{e}') from e



raise PluginInstallError(f'插件 {plugin} 依赖安装失败：{e}') from e 这个异常是不是不会处理？导致安装失败->启动失败->重启->安装失败的循环？

```
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
[INFO] Starting granian (main PID: 7694)
[INFO] Listening at: http://0.0.0.0:8000
[INFO] Spawning worker-1 with PID: 7695
2026-01-15 15:25:32.250 | INFO     | - | 检测插件依赖...
⠋ 安装插件 oauth2 依赖... 3/6 0:00:00
Process granian-worker:
Traceback (most recent call last):
  File "/fbapy/backend/plugin/requirements.py", line 65, in install_requirements
    subprocess.check_call(
    ~~~~~~~~~~~~~~~~~~~~~^
        pip_install,
        ^^^^^^^^^^^^
        stdout=subprocess.DEVNULL,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
        stderr=subprocess.DEVNULL,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/local/lib/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['uv', 'pip', 'install', '-r', PosixPath('/fbapy/backend/plugin/oauth2/requirements.txt'), '-i', 'https://mirrors.aliyun.com/pypi/simple/']' returned non-zero exit status 2.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.13/multiprocessing/process.py", line 313, in _bootstrap
    self.run()
    ~~~~~~~~^^
  File "/usr/local/lib/python3.13/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/granian/server/mp.py", line 73, in wrapped
    callback = callback_loader()
  File "/usr/local/lib/python3.13/site-packages/granian/_internal.py", line 65, in load_target
    module = load_module(path)
  File "/usr/local/lib/python3.13/site-packages/granian/_internal.py", line 47, in load_module
    __import__(module_name)
    ~~~~~~~~~~^^^^^^^^^^^^^
  File "/fbapy/backend/main.py", line 25, in <module>
    install_requirements(plugin)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/fbapy/backend/plugin/requirements.py", line 77, in install_requirements
    raise PluginInstallError(f'插件 {plugin} 依赖安装失败：{e}') from e
backend.plugin.requirements.PluginInstallError: 插件 oauth2 依赖安装失败：Command '['uv', 'pip', 'install', '-r', PosixPath('/fbapy/backend/plugin/oauth2/requirements.txt'), '-i', 'https://mirrors.aliyun.com/pypi/simple/']' returned non-zero exit status 2.
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
[INFO] Starting granian (main PID: 7715)
[INFO] Listening at: http://0.0.0.0:8000
[INFO] Spawning worker-1 with PID: 7716
2026-01-15 15:25:34.250 | INFO     | - | 检测插件依赖...
⠋ 安装插件 oauth2 依赖... 3/6 0:00:00
Process granian-worker:
Traceback (most recent call last):
  File "/fbapy/backend/plugin/requirements.py", line 65, in install_requirements
    subprocess.check_call(
    ~~~~~~~~~~~~~~~~~~~~~^
        pip_install,
        ^^^^^^^^^^^^
        stdout=subprocess.DEVNULL,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
        stderr=subprocess.DEVNULL,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/local/lib/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['uv', 'pip', 'install', '-r', PosixPath('/fbapy/backend/plugin/oauth2/requirements.txt'), '-i', 'https://mirrors.aliyun.com/pypi/simple/']' returned non-zero exit status 2.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.13/multiprocessing/process.py", line 313, in _bootstrap
    self.run()
    ~~~~~~~~^^
  File "/usr/local/lib/python3.13/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/granian/server/mp.py", line 73, in wrapped
    callback = callback_loader()
  File "/usr/local/lib/python3.13/site-packages/granian/_internal.py", line 65, in load_target
    module = load_module(path)
  File "/usr/local/lib/python3.13/site-packages/granian/_internal.py", line 47, in load_module
    __import__(module_name)
    ~~~~~~~~~~^^^^^^^^^^^^^
  File "/fbapy/backend/main.py", line 25, in <module>
    install_requirements(plugin)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/fbapy/backend/plugin/requirements.py", line 77, in install_requirements
    raise PluginInstallError(f'插件 {plugin} 依赖安装失败：{e}') from e
backend.plugin.requirements.PluginInstallError: 插件 oauth2 依赖安装失败：Command '['uv', 'pip', 'install', '-r', PosixPath('/fbapy/backend/plugin/oauth2/requirements.txt'), '-i', 'https://mirrors.aliyun.com/pypi/simple/']' returned non-zero exit status 2.
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
[INFO] Starting granian (main PID: 7736)
[INFO] Listening at: http://0.0.0.0:8000
[INFO] Spawning worker-1 with PID: 7737
2026-01-15 15:25:36.250 | INFO     | - | 检测插件依赖...
⠋ 安装插件 oauth2 依赖... 3/6 0:00:00
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -22,23 +22,29 @@ ENV UV_COMPILE_BYTECODE=1 \
 
 # Install dependencies with cache
 RUN --mount=type=cache,target=/root/.cache/uv \
-    uv sync --frozen --no-default-groups --group server
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --locked --no-default-groups --group server --no-install-project
 
 # === Runtime base server image ===
-FROM python:3.10-slim AS base_server
+FROM python:3.10-slim-bookworm AS base_server
 
 RUN sed -i 's/deb.debian.org/mirrors.ustc.edu.cn/g' /etc/apt/sources.list.d/debian.sources \
     && apt-get update \
-    && apt-get install -y --no-install-recommends supervisor \
+    && apt-get install -y --no-install-recommends curl ca-certificates supervisor \
     && rm -rf /var/lib/apt/lists/*
+
+ADD https://astral.sh/uv/install.sh /uv-installer.sh
+
+RUN sh /uv-installer.sh && rm /uv-installer.sh
+
+ENV PATH="/root/.local/bin/:$PATH"
 
 COPY --from=builder /fba /fba
 
 COPY --from=builder /usr/local /usr/local
 
 COPY deploy/backend/supervisor/supervisord.conf /etc/supervisor/supervisord.conf
-
-WORKDIR /fba/backend
 
 # === FastAPI server image ===
 FROM base_server AS fba_server

--- a/backend/plugin/requirements.py
+++ b/backend/plugin/requirements.py
@@ -1,5 +1,6 @@
 import os
 import subprocess
+import sys
 
 from importlib.metadata import PackageNotFoundError, distribution
 
@@ -23,6 +24,11 @@ def get_plugins() -> list[str]:
     from backend.plugin.core import get_plugins as _get_plugins
 
     return _get_plugins()
+
+
+def _is_in_virtualenv() -> bool:
+    """检测当前是否在虚拟环境中运行"""
+    return hasattr(sys, 'real_prefix') or (hasattr(sys, 'base_prefix') and sys.base_prefix != sys.prefix)
 
 
 def install_requirements(plugin: str | None) -> None:  # noqa: C901
@@ -56,6 +62,8 @@ def install_requirements(plugin: str | None) -> None:  # noqa: C901
         if missing_dependencies:
             try:
                 pip_install = ['uv', 'pip', 'install', '-r', requirements_file]
+                if not _is_in_virtualenv():
+                    pip_install.append('--system')
                 if settings.PLUGIN_PIP_CHINA:
                     pip_install.extend(['-i', settings.PLUGIN_PIP_INDEX_URL])
 
@@ -91,6 +99,8 @@ def uninstall_requirements(plugin: str) -> None:
     if os.path.exists(requirements_file):
         try:
             pip_uninstall = ['uv', 'pip', 'uninstall', '-r', str(requirements_file), '-y']
+            if not _is_in_virtualenv():
+                pip_uninstall.append('--system')
             subprocess.check_call(pip_uninstall, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
         except subprocess.CalledProcessError as e:
             raise PluginInstallError(f'插件 {plugin} 依赖卸载失败：{e}') from e

